### DISCUSSION TITLE
Integrating the vaccinated Population

### DISCUSSION THREAD

**Discussion by krageth** [epiforecasts/EpiNow2#318] [Q&A]
This is definitely not a 'bug' or 'issue' but more of a discussion (but I did not find a discussion string anywhere, my apologies if there is one), but how best should we begin accounting for the vaccinated population. Assuming we have vaccination release plans and the associated vaccinated population time series, how best do we go about accounting for those that are vaccinated as we (hopefully) move toward herd immunity in the coming months?

**Comment by seabbs** [comment]
Hi @krageth 

Apologies for the delay on getting back to you on this I paused development to work on some other projections that were time critical but should be back on this shortly. 

This is a very pertinent question and as of yet we don't have support  for this in forecasts though it is naturally included in the non-parametric estimate of Rt etc when nowcasting.  If you wanted to back out the Rt you would estimate if people had not been vaccinated that is a transform of the returned estimates but as I read your question is not what you are interested in. 

For short term forecasting needs this is probably not going to have a major impact but I agree its a feature we should add. 

For longer term projections it is definitely an issue. Here we currently have a non-optimal solution which would be to project Rt forwards in time in the presence of no vaccination, then scale the posterior Rt estimates based on the expected impact of vaccination (which depends on who is being vaccinated). To then get back to a case forecast the adjusted Rt samples can be fed into `simulate_infections` along with the model that was used to estimate them (using the GitHub version of the package). 

To implement this in the most basic sense we would need to allow the user to pass in a time-varying susceptible population (so that those who have been vaccinated can be removed) which should be a fairly small change. My concern with this approach is that all vaccine programs currently are very targeted and so perhaps a more nuanced method is needed. 

Sorry quite a bit to unpack there. I will think more on this and circle back.

Just a note discussion is here for future qs but no problem continuing in this issue: https://github.com/epiforecasts/EpiNow2/discussions

Sam

**Comment by krageth** [comment]
@seabbs Thank you for your feedback, I greatly appreciate it!
